ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/V1/Snapshots.hs

```diff
@@ -0,0 +1,255 @@
+         snapshot and start again.
+
+       In case we found an snapshot, we will overwrite (either literally
+       overwriting it or using some feature from the specific backend used) the
+       @BackingStore@ tables with the contents from @tables@ from said snapshot
+       as it was left in whatever state it was when the node shut down.
+
+  3. The deserialized ledger state and tables will be then used as the initial
+       ledger state for the ledger database.
+
+  4. Reapply the immutable blocks after the snapshot to obtain the ledger state
+       at the immutable tip. The blocks to reapply are streamed from the Immutable
+       DB, using an iterator.
+
+       Note that we can /reapply/ these blocks, which is quicker than applying
+       them, as the existence of a snapshot newer than these blocks proves (unless
+       the on-disk database has been tampered with, but this is not an attack we
+       intend to protect against, as this would mean the machine has already been
+       compromised) that they have been successfully applied in the past.
+
+  Reading and applying blocks is costly. Typically, very few blocks need to be
+  reapplied in practice. However, there is one exception: when the serialisation
+  format of the ledger state changes, all snapshots (written using the old
+  serialisation format) will fail to deserialise, and all blocks starting from
+  genesis will have to be reapplied.
+
+  At this point, the node carries a @DbChangelog@ that is initialized and ready
+  to be applied blocks on the volatile database.
+
+  == Taking snapshots during normal operation
+
+  Snapshots are taken by the @'copyAndSnapshotRunner'@ when the disk policy
+  dictates to do so. Whenever the chain grows past @k@ blocks, said runner will
+  copy the blocks which are more than @k@ blocks from the tip (i.e. the ones
+  that must be considered immutable) to the immutable database and then:
+
+  1. Every time we have processed a specific amount of blocks since the last
+       flush (set by default to 100), perform a flush of differences in the
+       'DbChangelog' up to the immutable db tip.
+
+  2. If dictated by the disk policy, flush immediately all the differences up to
+       the immutable db tip and serialize (using 'EncodeDisk') the DbChangelog
+       in-memory ledger states anchor (@ExtLedgerState blk EmptyMK@).
+
+       A directory is created named after the slot number of the ledger state
+       being snapshotted, and the serialization from above is written into the
+       @\<slotNumber\>/state@ file and the @BackingStore@ tables are copied into
+       the @\<slotNumber\>/tables@ file.
+
+  3. There is a maximum number of snapshots that should exist in the disk at any
+     time, dictated by the @DiskPolicy@, so if needed, we will trim out old
+     snapshots.
+
+  == Flush during startup and snapshot at the end of startup
+
+  Due to the nature of the database having to carry around all the differences
```

Comment:
When I read "the database", I think of the BackingStore before I think of the LedgerDB. So I suggest changing it to "the V1 LedgerDB having to carry..."